### PR TITLE
feat: biw smart item compatibility with builder

### PR DESCRIPTION
**Pull request by D4rWiNSS**
# What? 

This PR made builder in world able to open builder scenes with smart items and adds a way to don't lose the information of Builder about smart items when travelling to the renderer.

# Why? 

Currently, the builder scenes with smart items can't be opened in the builder in world.

Also, This will make the compatibility between builder and builder in world even more robust since you won't lose your smart item data when you open the scene in builder in world

### CODE REVIEW

File: kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts [hunk 1]
```diff
@@ -67,12 +68,21 @@ export async function toBuilderFromStateDefinitionFormat(
       builderComponents[builderComponent.id] = builderComponent
     }
 
+    let entityname = entityId
```
Comment (kuruk-mm) [inline kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts]:
```suggestion
    let entityName = entityId
```

File: kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts [hunk 2]
```diff
@@ -67,12 +68,21 @@ export async function toBuilderFromStateDefinitionFormat(
...
+
+    // We iterate over the name of the entities to asign it in a builder format
+    for (let component of Object.values(builderComponents)) {
+      if (component.type === 'Name') {
+        entityname = component.data.builderValue
```
Comment (kuruk-mm) [inline kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts]:
```suggestion
        entityName = component.data.builderValue
```

File: kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts [hunk 3]
```diff
@@ -67,12 +68,21 @@ export async function toBuilderFromStateDefinitionFormat(
...
+      }
+    }
+
     // we add the entity to builder format
     let builderEntity: BuilderEntity = {
       id: entityId,
       components: builderComponentsIds,
       disableGizmos: false,
-      name: entityId
+      name: entityname
```
Comment (kuruk-mm) [inline kernel/packages/shared/apis/SceneStateStorageController/StorableSceneStateTranslation.ts]:
```suggestion
      name: entityName
```

**Review by kuruk-mm — CHANGES_REQUESTED** [review]

**Review by kuruk-mm — APPROVED** [review]
The code looks good. I didn't test it.

**Review by sandrade-dcl — APPROVED** [review]
LGTM!

**Pull request merged**